### PR TITLE
Adam integration with grouped layers

### PR DESCRIPTION
**Pull request by JorgeMartinezG**
This PR includes WFP ADAM layers into PRISM.

## **Changes**
- Added field group on layers type schema.
- Included adam earthquake layer.
- Included adam tropical storms layers, displayed as single layer.
- Modified date parser to remove time values.

image

You can see demo of this PR here: http://homeless-giraffe.surge.sh/

### CODE REVIEW

**Comment by ericboucher** [comment]
When clicking on a past month, we get the following errors:

Screen Shot 2021-04-05 at 4 04 50 PM

Are they avoidable? Could we directly select a valid date?

**Comment by JorgeMartinezG** [comment]
@ericboucher Fixed issue. Basically it was popping alerts for each layer. Just modified the group schema which specified if a layer is the main one. Branch was rebased wrt master.

Also, used a subset of the global layer to be consistent with a given country. For this example, I used mozambique. You can see it within the demo page: http://homeless-giraffe.surge.sh/

**Comment by dubwalla** [comment]
Looks good, thanks Jorge. One thing I'm seeing is that when you click on a month, it requests a layer for the first day of the month. If a tropical storm happens on a day besides the first of the month, clicking on a month results in the mismatched date popup alert. It would be good if interaction by month fetches whatever the first event within a month is instead of just the first of the month. I'll put this into a separate issue

**Comment by dubwalla** [comment]
Added issue https://github.com/WFP-VAM/prism-frontend/issues/153 for the date mismatch error @ericboucher 
 @JorgeMartinezG

**Comment by ericboucher** [comment]
That is already the expected behavior if I recall correctly so something else must be happening here. Or maybe it's because we select multiple layers at once and maybe their dates are not matching?

**Comment by JorgeMartinezG** [comment]
Master has the same behaviour. the logic for this is set here: https://github.com/WFP-VAM/prism-frontend/blob/master/src/components/MapView/index.tsx#L148

Related to ADAM tropical storms layers, tracks and wind buffers are generated from nodes layers, using the same date.

**Comment by ericboucher** [comment]
Hmm I think you're right, maybe it's just because I am used to the Mongolia version which has all layers with data on the first

https://prism-mongolia.org/

File: src/context/mapStateSlice/index.ts [hunk 1]
```diff
@@ -41,10 +42,34 @@ export const mapStateSlice = createSlice({
   name: 'mapState',
   initialState,
   reducers: {
-    addLayer: ({ layers, ...rest }, { payload }: PayloadAction<LayerType>) => ({
-      ...rest,
-      layers: layers.filter(layer => keepLayer(layer, payload)).concat(payload),
-    }),
+    addLayer: ({ layers, ...rest }, { payload }: PayloadAction<LayerType>) => {
+      // Validate if layer belongs to a group.
```
Comment (ericboucher) [inline src/context/mapStateSlice/index.ts:46]: Validate -> Check if a layer

File: src/context/mapStateSlice/index.ts [hunk 2]
```diff
@@ -41,10 +42,34 @@ export const mapStateSlice = createSlice({
...
+          layers: layers
+            .filter(layer => keepLayer(layer, payload))
+            .concat(payload),
+        };
+      }
+
+      const { name } = payload.group;
+      // Get all layers that belong to a group.
+      const items = Object.values(LayerDefinitions).filter(l => {
+        const groupName = l.group ? l.group.name : undefined;
```
Comment (ericboucher) [inline src/context/mapStateSlice/index.ts:59]: You could simply do a one line:

`l.group?.name === name`

**Review by ericboucher — APPROVED** [review]
Just a few cleanups, looks good!

File: src/context/mapStateSlice/index.ts [hunk 3]
```diff
@@ -41,10 +42,32 @@ export const mapStateSlice = createSlice({
...
+      }
+
+      const { name } = payload.group;
+      // Get all layers that belong to a group.
+      const items = Object.values(LayerDefinitions).filter(l => {
+        if (l.group?.name === name) {
+          return true;
+        }
+
+        return false;
```
Comment (ericboucher) [inline src/context/mapStateSlice/index.ts:63]: I meant that you can go even one step further and just:

`Object.values(LayerDefinitions).filter(l => (l.group?.name === name))`

File: src/context/mapStateSlice/index.ts [hunk 4]
```diff
@@ -41,10 +42,32 @@ export const mapStateSlice = createSlice({
...
+          ...rest,
+          layers: layers
+            .filter(layer => keepLayer(layer, payload))
+            .concat(payload),
+        };
+      }
+
+      const { name } = payload.group;
+      // Get all layers that belong to a group.
+      const items = Object.values(LayerDefinitions).filter(l => {
```
Comment (ericboucher) [inline src/context/mapStateSlice/index.ts:58]: maybe rename `items` to `groupedLayer` or similar?

**Pull request merged**